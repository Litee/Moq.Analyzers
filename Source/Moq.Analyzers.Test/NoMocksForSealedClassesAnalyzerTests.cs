using ApprovalTests;
using ApprovalTests.Reporters;
using Microsoft.CodeAnalysis.Diagnostics;
using System.IO;
using TestHelper;
using Xunit;

namespace Moq.Analyzers.Test
{
    [UseReporter(typeof(DiffReporter))]
    public class NoMocksForSealedClassesAnalyzerTests : DiagnosticVerifier
    {

        [Fact]
        public void ShouldFailIfFileIsSealed()
        {
            Approvals.Verify(VerifyCSharpDiagnostic(File.ReadAllText("Data/MockSealedClass.cs")));
        }

        protected override DiagnosticAnalyzer GetCSharpDiagnosticAnalyzer()
        {
            return new NoMocksForSealedClassesAnalyzer();
        }
    }
}