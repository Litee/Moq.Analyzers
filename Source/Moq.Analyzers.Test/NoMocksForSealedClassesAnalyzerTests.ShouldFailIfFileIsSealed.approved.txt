Diagnostic 1
	Id: Moq1000
	Location: SourceFile(Test0.cs[220..229))
	Code: MyService
	Severity: Warning
	Message: Sealed classes cannot be mocked.

Diagnostic 2
	Id: Moq1000
	Location: SourceFile(Test0.cs[268..277))
	Code: MyService
	Severity: Warning
	Message: Sealed classes cannot be mocked.

Diagnostic 3
	Id: Moq1000
	Location: SourceFile(Test0.cs[316..341))
	Code: MockSealedClass.MyService
	Severity: Warning
	Message: Sealed classes cannot be mocked.

Diagnostic 4
	Id: Moq1000
	Location: SourceFile(Test0.cs[384..409))
	Code: MockSealedClass.MyService
	Severity: Warning
	Message: Sealed classes cannot be mocked.

